include/irr/asset/ICPUDescriptorSet.h

```diff
@@ -193,6 +193,41 @@ class ICPUDescriptorSet final : public IDescriptorSet<ICPUDescriptorSetLayout>,
 			return m_descriptors->size();
 		}
 
+	private:
+		void restoreFromDummy_impl(IAsset* _other, uint32_t _levelsBelow) override
+		{
+			auto* other = static_cast<ICPUDescriptorSet*>(_other);
+
+			if (_levelsBelow)
+			{
+				--_levelsBelow;
+				m_layout->restoreFromDummy(other->getLayout(), _levelsBelow);
+				for (auto it = m_descriptors->begin(); it != m_descriptors->end(); it++)
+				{
+					auto descriptor = it->desc.get();
+					if (!descriptor)
+						continue;
+					const auto i = it - m_descriptors->begin();
+					auto* d_other = other->m_descriptors->begin()[i].desc.get();
+
+					switch (descriptor->getTypeCategory())
+					{
+					case IDescriptor::EC_BUFFER:
+						static_cast<asset::ICPUBuffer*>(descriptor)->restoreFromDummy(static_cast<ICPUBuffer*>(d_other), _levelsBelow);
+						break;
+					case IDescriptor::EC_IMAGE:
+						static_cast<asset::ICPUImageView*>(descriptor)->restoreFromDummy(static_cast<ICPUImageView*>(d_other), _levelsBelow);
+						if (descriptor->getTypeCategory() == IDescriptor::EC_IMAGE && it->image.sampler)
+							it->image.sampler->restoreFromDummy(other->m_descriptors->begin()[i].image.sampler.get(), _levelsBelow);
+						break;
+					case IDescriptor::EC_BUFFER_VIEW:
+						static_cast<asset::ICPUBufferView*>(descriptor)->restoreFromDummy(static_cast<ICPUBufferView*>(d_other), _levelsBelow);
```

Comment:
if we'll already check if we can restore entire dependency tree, then this should call `restoreFromDummy_impl` directly instead of `restoreFromDummy` for dependencies

Tied up with the same thread https://github.com/buildaworldnet/IrrlichtBAW/pull/549#discussion_r507664092